### PR TITLE
Make use of unstructured when using SSA to prevent default values causing issues

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

There have been some issues in the past when we tested SSA in the current implementation with default values and it's also a known issue with the controller-runtime, e.g.: https://github.com/kubernetes-sigs/controller-runtime/issues/1669 (only to link one issue).

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)

## Discussion

This will add some more cpu cycles to the operator when pushing the patch but I think this is the simplest way to approach this issue right now. Once the controller-runtime has a better approach we can remove this workaround.

## Testing

Ran a manual e2e test.

CI will run all e2e tests with the SSA feature as I enabled it in the operator deployment.

## Documentation

-

## Follow-up

-

### CODE REVIEW

**Review by nicmorales9 — APPROVED** [review]
overall looks good if it works in testing, though it'd be good to get clarification on the forceOwnership thread.

File: controllers/cluster_controller.go [hunk 1]
```diff
@@ -442,7 +448,22 @@ func (r *FoundationDBClusterReconciler) updateOrApply(ctx context.Context, clust
...
+			return err
+		}
+
+		unstructuredPatch := &unstructured.Unstructured{}
+		_, _, err = r.decodingSerializer.Decode(outBytes, nil, unstructuredPatch)
+		if err != nil {
+			return err
+		}
+
+		return r.Status().Patch(ctx, unstructuredPatch, client.Apply, client.FieldOwner("fdb-operator")) //, client.ForceOwnership)
```
[the remark this answers is not in the document]
Comment (johscheuer) [reply controllers/cluster_controller.go:539]: This doesn't work for sub resources like the status.

File: e2e/fixtures/fdb_operator_client.go [hunk 2]
```diff
@@ -466,7 +466,7 @@ spec:
           - --max-concurrent-reconciles=5
           - --zap-log-level=debug
           - --minimum-required-uptime-for-cc-bounce=60s
-          #- --server-side-apply
+          - --server-side-apply
```
Comment (johscheuer) [inline e2e/fixtures/fdb_operator_client.go:469]: This should be a feature that can be enabled/disabled.

**Review by nicmorales9 — APPROVED** [review]
looks right to me

File: e2e/fixtures/options.go [hunk 3]
```diff
@@ -192,6 +193,12 @@ func (options *FactoryOptions) BindFlags(fs *flag.FlagSet) {
 		true,
 		"defines if the operator tests should print the state of the cluster and the according Pods for better debugging.",
 	)
+	fs.BoolVar(
+		&options.featureOperatorServerSideApply,
+		"feature-server-side-apply",
+		true, // TODO change to false.
```
Comment (johscheuer) [inline e2e/fixtures/options.go:199]:
```suggestion
		false,
```

**Pull request merged**